Mage/src/main/java/mage/abilities/effects/ContinuousEffects.java

```diff
@@ -1286,6 +1315,15 @@ public synchronized void addEffect(ContinuousEffect effect, Ability source) {
             case PREVENTION:
                 preventionEffects.addEffect((PreventionEffect) effect, source);
                 break;
+            case EVASION:
+                evasionEffects.addEffect((EvasionEffect) effect, source);
+                // Some evasion effects are using the layer system to alter
+                // the evading permanent. They are listed as both evasionEffects,
+                // and layeredEffects.
+                if (effect.getLayer() == Layer.RulesEffects) {
+                    layeredEffects.addEffect(effect, source);
```

Comment:
This is my current solution to have the evasion of Bristling Boar or Gorilla Berserkers to apply on the `RulesEffect` layer, in addition to being evasion effects to be collected in `getRules`.

Is there any reason to not have effects in two different lists?

Reply:
It's definitely the wrong use of effects. No duplications.